clients/da-vinci-client/src/main/java/com/linkedin/davinci/blobtransfer/server/P2PFileTransferServerHandler.java

```diff
@@ -101,9 +107,24 @@ protected void channelRead0(ChannelHandlerContext ctx, FullHttpRequest httpReque
     }
     final BlobTransferPayload blobTransferRequest;
     final File snapshotDir;
+    boolean isHybridStore;
+
     try {
       blobTransferRequest = parseBlobTransferPayload(URI.create(httpRequest.uri()));
       snapshotDir = new File(blobTransferRequest.getSnapshotDir());
+      isHybridStore = blobSnapshotManager.isStoreHybrid(blobTransferRequest.getStoreName());
+
+      // recreate snapshot for hybrid store
+      if (isHybridStore) {
+        try {
+          BlobTransferPartitionMetadata metadataForHybrid = prepareMetadata(blobTransferRequest, ctx);
+          blobSnapshotManager.recreateSnapshotForHybrid(blobTransferRequest, metadataForHybrid);
```

Comment:
per our offline discussion, we want to do a bit of refactoring so `blobSnapshotManager` would handle metadata creation, lifecycle management, snapshot creation etc., so server handler would just call `blobSnapshotManager` to obtain all of these to make codes more readable and easy to maintain.